### DISCUSSION TITLE
[Help] Using consult-fd to Search and Open Files in Emacs on Windows

### DISCUSSION THREAD

**Discussion by Ypot** [minad/consult#1071] [Q&A] [answered]
Hi

Consult-fd it's the only way I've found in Emacs that works, within Windows, to search for files within subdirectories. And it's also fast!

The problem I have is that when I open the files, I don't have control over which program they are open with. For example, I would like to open PDF files with an external program, but they're open within Emacs.

Is it possible to use the default system program to open files from the consult-fd search results?

Thanks a lot for these packages.
Best regards

**Comment by Ypot** [comment]
I have tried this, without any result:

```
(setq consult-file-handler-alist
      '(("\\.pdf\\'" . system)))
```

**Comment by minad** [comment, marked as answer]
Consult relies on the normal `find-file` to open the file. It is out of scope of Consult to provide a different file handler mechanism. Note that you can use Embark to open files externally. See the `embark-open-externally` action which is bound to the key `x` in the Embark file map.
